Content.Shared/Bed/Sleep/SleepingSystem.cs

```diff
@@ -328,4 +336,17 @@ public sealed partial class WakeActionEvent : InstantActionEvent;
 /// Raised on an entity when they fall asleep or wake up.
 /// </summary>
 [ByRefEvent]
-public record struct SleepStateChangedEvent(bool FellAsleep);
+public sealed class SleepStateChangedEvent : EntityEventArgs
+{
+    public bool FellAsleep = false;
+
+    /// <summary>
+    ///     The amount of time this entity slept for. Null if <see cref="FellAsleep"/> is true.
+    /// </summary>
+    public TimeSpan? TimeSlept;
+
+    public SleepStateChangedEvent(bool fellAsleep)
+    {
+        FellAsleep = fellAsleep;
+    }
+}
```

Comment:
dont do this, add `, TimeSpan? TimeSlept = null` and instead of setting .TimeSlept = ... pass it as extra param
then add deltav comment on the line